src/solver/utils/mps_utils.cpp

```diff
@@ -514,16 +521,18 @@ mpsWriterFactory::mpsWriterFactory(
 
 bool mpsWriterFactory::doWeExportMPS()
 {
-    // Current optimization number is not involved yet in the return value
-    // but will be soon
-    return export_mps_;
+    if (export_mps_ == Data::mpsExportStatus::EXPORT_BOTH_OPTIMS)
+        return true;
+    if ((int)export_mps_ == current_optim_number_)
+        return true;
+    return false;
```

Comment:
Return false if NO_EXPORT

Reply:
Also, avoid converting `enums` to `int`, expand this `if (export_mps_ == FIST_OPTIM && current_optim_number_ == 1)` etc.

Reply:
> Return false if NO_EXPORT

What for ? If **NO_EXPORT**, this function returns already **false**

Reply:
> Also, avoid converting `enums` to `int`, expand this `if (export_mps_ == FIST_OPTIM && current_optim_number_ == 1)` etc.

To convert an enum class value into an int, I now use `static_cast<int>(...)`

Reply:
It would be best not to convert